container_pipeline/start_build.py

```diff
@@ -0,0 +1,75 @@
+import json
+import logging
+import os
+import sys
+
+import container_pipeline.utils as utils
+from container_pipeline.lib import settings
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.lib.queue import JobQueue
+
+
+def create_project(appid, jobid, repo_url, repo_branch, repo_build_path, target_file, notify_email, desired_tag, depends_on, test_tag):
+    job_name = utils.get_job_name({
+        'appid': appid, 'jobid': jobid, 'desired_tag': desired_tag})
+    project = utils.get_job_hash(job_name)
+    is_openshift_good = True
+
+    openshift = Openshift(logger=logger)
+    try:
+        openshift.login("test-admin", "test")
+        openshift.create(project)
+        openshift.clean_project(project)
+    except OpenshiftError:
+        is_openshift_good = False
+        return 1
+
+    try:
+        template_path = os.path.join(
+            os.path.dirname(__file__), 'template.json')
+        openshift.upload_template(project, template_path, {
+            'SOURCE_REPOSITORY_URL': repo_url,
+            'REPO_BRANCH': repo_branch,
+            'APPID': appid,
+            'JOBID': jobid,
+            'REPO_BUILD_PATH': repo_build_path,
+            'TARGET_FILE': target_file,
+            'NOTIFY_EMAIL': notify_email,
+            'DESIRED_TAG': desired_tag,
+            'TEST_TAG': test_tag})
+    except OpenshiftError:
+        is_openshift_good = False
+        return 1
+
+    if(is_openshift_good):
```

Comment:
in what scenario `is_openshift_good` would be `False`? all the other places we are setting it to `False` and returning.